libs/libc/wchar/lib_wcrtomb.c

```diff
@@ -57,18 +51,37 @@
 
 size_t wcrtomb(FAR char *s, wchar_t wc, FAR mbstate_t *ps)
 {
-  int retval = 0;
-  char buf[MB_LEN_MAX];
-
   if (s == NULL)
     {
-      retval = wctomb(buf, wc);
+      return 0;
+    }
+  else if ((unsigned)wc < 0x80)
+    {
+      *s = wc;
+      return 1;
+    }
+  else if ((unsigned)wc < 0x800)
+    {
+      *s++ = 0xc0 | (wc >> 6);
+      *s = 0x80 | (wc & 0x3f);
+      return 2;
+    }
+  else if ((unsigned)wc < 0xd800 || (unsigned)wc - 0xe000 < 0x2000)
+    {
+      *s++ = 0xe0 | (wc >> 12);
+      *s++ = 0x80 | ((wc >> 6) & 0x3f);
+      *s = 0x80 | (wc & 0x3f);
+      return 3;
     }
-  else
+  else if ((unsigned)wc - 0x10000 < 0x100000)
```

Comment:
So here we assume that `int` is at lease 32bit? Maybe switching to fixed sized `uint32_t` or at least to `unsigned long` is better? Also logic can be reworked to avoid subtraction of unsigned: `else if ((unsigned)wc < 0x110000)`

Reply:
this is still not answered.

Reply:
Done! Sorry, i forgot.